### PR TITLE
Add ConfChange::AddLearnerNode command.

### PR DESCRIPTION
**Pull request by hicqu**
This PR is for support raft learner. See https://github.com/coreos/etcd/pull/8751

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @overvenus @BusyJay

**Comment by ngaut** [comment]
LGTM

**Comment by hicqu** [comment]
@BusyJay  PTAL thanks.

File: proto/pdpb.proto [hunk 1]
```diff
@@ -233,6 +233,7 @@ message RegionHeartbeatRequest {
 enum ConfChangeType {
     AddNode    = 0;
     RemoveNode = 1;
+    AddLearnerNode = 2;
```
Comment (BusyJay) [inline proto/pdpb.proto]: Why change this?

**Comment by BusyJay** [comment]
Please add pr description to show what the changes come from.

File: proto/pdpb.proto (hunk 1 above)
Comment (hicqu) [reply proto/pdpb.proto]: I thought pd should also support `add learner node` command.

Comment (BusyJay) [reply proto/pdpb.proto]: Yes. But not in this pr.

**Comment by hicqu** [comment]
@BusyJay PTAL thanks.

File: proto/eraftpb.proto [hunk 2]
```diff
@@ -72,11 +72,13 @@ message HardState {
 
 message ConfState {
     repeated uint64 nodes = 1;
+    repeated uint64 learner_nodes = 2;
```
Comment (overvenus) [inline proto/eraftpb.proto]: Why not `learners`[0]?

[0] https://github.com/coreos/etcd/pull/8751/files#diff-ed53c23ad570cdd6dce39cbd1b751cacR80

File: proto/metapb.proto [hunk 3]
```diff
@@ -51,6 +51,7 @@ message Region {
     optional bytes  end_key             = 3;
     optional RegionEpoch region_epoch   = 4;
     repeated Peer   peers               = 5;
+    repeated Peer   learners		= 6;
```
Comment (BusyJay) [inline proto/metapb.proto]: Should remove this too.

**Comment by hicqu** [comment]
@BusyJay @overvenus PTAL thanks.

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**